app/com/linkedin/drelephant/spark/SparkMetricsAggregator.scala

```diff
@@ -78,27 +83,32 @@ class SparkMetricsAggregator(private val aggregatorConfigurationData: Aggregator
   }
 
   //calculates the resource usage by summing up the resources used per executor
-  private def calculateResourceUsage(executorSummaries: Seq[ExecutorSummary], executorMemoryBytes: Long): (BigInt, BigInt) = {
+  private def calculateResourceUsage(data: SparkApplicationData): (BigInt, BigInt) = {
+    val executorSummaries = data.executorSummaries
     var sumResourceUsage: BigInt = 0
-    var sumResourcesAllocatedForUse : BigInt = 0
+    var sumResourcesAllocatedForUse: BigInt = 0
     executorSummaries.foreach(
       executorSummary => {
-        var memUsedBytes: Long = executorSummary.peakJvmUsedMemory.getOrElse(JVM_USED_MEMORY, 0).asInstanceOf[Number].longValue + MemoryFormatUtils.stringToBytes(SPARK_RESERVED_MEMORY)
-        var timeSpent: Long = executorSummary.totalDuration
-        val bytesMillisUsed = BigInt(memUsedBytes) * timeSpent
-        val bytesMillisAllocated = BigInt(executorMemoryBytes) * timeSpent
+        val memoryOverhead = overheadMemoryBytesOf(data).get
+        val roundedContainerBytes = getRoundedContainerBytes(data).get
+        val memUsedBytes: Long = executorSummary.peakJvmUsedMemory.getOrElse(JVM_USED_MEMORY, 0).asInstanceOf[Number].longValue + MemoryFormatUtils.stringToBytes(SPARK_RESERVED_MEMORY) + memoryOverhead
```

Comment:
As discussed SPARK_RESERVED_MEMORY is part of JVM_USED_MEMORY , are u not double counting it here